src/feeds/InvPriceFeed.sol

```diff
@@ -0,0 +1,156 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.4;
+
+interface IChainlinkFeed {
+    function aggregator() external view returns (address aggregator);
+
+    function decimals() external view returns (uint8);
+
+    function latestRoundData()
+        external
+        view
+        returns (
+            uint80 roundId,
+            int256 invDollarPrice,
+            uint256 startedAt,
+            uint256 updatedAt,
+            uint80 answeredInRound
+        );
+}
+
+interface IAggregator {
+    function maxAnswer() external view returns (int192);
+
+    function minAnswer() external view returns (int192);
+}
+
+interface ICurvePool {
+    function price_oracle(uint256 k) external view returns (uint256);
+}
+
+contract InvPriceFeed {
+    ICurvePool public constant tricrypto =
+        ICurvePool(0x5426178799ee0a0181A89b4f57eFddfAb49941Ec);
+
+    IChainlinkFeed public constant usdcToUsd =
+        IChainlinkFeed(0x8fFfFfd4AfB6115b954Bd326cbe7B4BA576818f6);
+
+    IChainlinkFeed public constant ethToUsd =
+        IChainlinkFeed(0x5f4eC3Df9cbd43714FE2740f5E3616155c5b8419);
+
+    uint256 public constant ethK = 0;
+    uint256 public constant invK = 1;
+
+    /**
+     * @notice Retrieves the latest round data for the INV token price feed
+     * @dev This function calculates the INV price in USD by combining the USDC to USD price from a Chainlink oracle and the INV to USDC ratio from the tricrypto pool
+     * @return roundId The round ID of the Chainlink price feed
+     * @return usdcUsdPrice The latest USDC price in USD computed from the INV/USDC and USDC/USD feeds
+     * @return startedAt The timestamp when the latest round of Chainlink price feed started
+     * @return updatedAt The timestamp when the latest round of Chainlink price feed was updated
+     * @return answeredInRound The round ID in which the answer was computed
+     */
+    function latestRoundData()
+        external
+        view
+        returns (uint80, int256, uint256, uint256, uint80)
+    {
+        return _latestRoundData();
+    }
+
+    /** 
+    @notice Retrieves the latest price for the INV token
+    @return price The latest price for the INV token
+    */
+    function latestAnswer() external view returns (int256) {
+        (, int256 price, , , ) = _latestRoundData();
+        return price;
+    }
+
```

Comment:
Any reason you prefer to have `_latestRoundData()` be an internal function, rather than just making `latestRoundData()` `public` instead of `external` and using it in `latestAnswer`?

Reply:
Fixed in [863bbd2](https://github.com/InverseFinance/FiRM/pull/58/commits/863bbd23cbfd871dacf8c690cd21a1285fd941ba)